nix/rollup-bridge-contracts.nix

```diff
@@ -25,26 +26,74 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [
     nodejs
     npmHooks.npmConfigHook
+    pkgs.foundry
+    pkgs.solc
+    pkgs.bash
   ];
 
   soljson26 = builtins.fetchurl {
-    url = "https://binaries.soliditylang.org/wasm/soljson-v0.8.26+commit.8a97fa7a.js";
-    sha256 = "1mhww44ni55yfcyn4hjql2hwnvag40p78kac7jjw2g2jdwwyb1fv";
+    url = "https://binaries.soliditylang.org/wasm/soljson-v0.8.28+commit.7893614a.js";
+    sha256 = "0ip1kafi7l5zkn69zj5c41al7s947wqr8llf08q33565dq55ivvj";
+  };
+
+  forgeStd = pkgs.fetchzip {
+    url = "https://github.com/foundry-rs/forge-std/archive/refs/tags/v1.9.6.zip";
+    sha256 = "sha256-4y1Hf0Te2oJxwKBOgVBEHZeKYt7hs+wTgdIO+rItj0E=";
+  };
+
+  solmate = pkgs.fetchFromGitHub {
+    owner = "transmissions11";
+    repo = "solmate";
+    rev = "c93f7716c9909175d45f6ef80a34a650e2d24e56";
+    sha256 = "sha256-zv8Jzap34N5lFVZV/zoT/fk73pSLP/eY427Go3QQM/Y="; # Replace with actual hash
+  };
+
+  dsTest = pkgs.fetchFromGitHub {
+    owner = "dapphub";
+    repo = "ds-test";
+    rev = "e282159d5170298eb2455a6c05280ab5a73a4ef0";
+    sha256 = "sha256-wXtNq4ZUohndNGs9VttOI9m9VW5QlVKOPtR8+mv2fBM=";
   };
 
   buildPhase = ''
     echo "Installing soljson"
     (cd create-nil-hardhat-project; bash install_soljson.sh ${soljson26})
     export BIOME_BINARY=${biome}/bin/biome
 
+    echo "Versions:"
+    solc --version
+    forge --version
+    cast --version
+
     cd rollup-bridge-contracts
-    pwd
     cp .env.example .env
 
-    echo "start compiling"
+    echo "Start Hardhat compiling:"
     npx hardhat clean && npx hardhat compile
   '';
 
+  doCheck = enableTesting;
+  checkPhase = ''
+    source .env
+
+    export FOUNDRY_ROOT=$(realpath ../)
+    solc=pkgs.solc;
```

Comment:
What does `solc=pkgs.solc` do? You either meant `solc="${pkgs.solc}` or something. But anyway, if `solc` is in your nativeBuildInputs - it's already in the `$PATH`